services/galley/test/integration/API.hs

```diff
@@ -272,7 +272,17 @@ tests s =
           -- As a lot of these tests are waiting on specific notifications to come through in a specified
           -- order, these tests will cause them to fail.
           -- See the Tasty docs on patterns. https://hackage.haskell.org/package/tasty-1.4.3#patterns
-          after AllFinish "$0 !~ /delete federation notifications/" $ test s "delete federation notifications" API.testDefederationNotifications
+          after AllFinish "$0 !~ /federation notifications/" $
+            testGroup
+              "federation notifications"
+              -- Run these tests in order by having them wait on each other.
+              -- The names need to be distint enough so that there isn't a loop with the regexes
+              [ test s "delete federation notifications" API.testDefederationNotifications,
+                after AllFinish "$0 ~ /delete federation notifications/" $ test s "connection removed notifications normal" API.testConnectionRemovedNotifications,
```

Comment:
I would remove the `API` qualifier here and below, too.

Reply:
Removed.